src/gsym.rs

```diff
@@ -71,6 +74,75 @@ impl SymResolver for GsymResolver {
         None
     }
 
+    fn find_line_info(&self, addr: u64) -> Option<AddressLineInfo> {
+        let addr = addr - self.loaded_address;
+        let idx = find_address(&self.ctx, addr);
+        let symaddr = self.ctx.addr_at(idx);
+        if addr < symaddr {
+            return None;
+        }
+        let addrinfo = self.ctx.addr_info(idx);
+        if addr >= (symaddr + addrinfo.size as u64) {
+            return None;
+        }
+
+        let addrdatas = parse_address_data(addrinfo.data);
+        for addrdata in addrdatas {
+            if addrdata.typ != InfoTypeLineTableInfo {
+                continue;
+            }
+            let lthdr = parse_line_table_header(addrdata.data);
+            if lthdr.is_err() {
+                #[cfg(debug_assertions)]
+                eprintln!("invalid line table header");
+                return None;
+            }
+            let (lthdr, bytes) = lthdr.unwrap();
+            let ops = &addrdata.data[bytes..];
+            let mut ltr = line_table_row_from(&lthdr, symaddr);
+            let mut saved_ltr = ltr.clone();
+            let mut row_cnt = 0;
+            let mut pc = 0;
+            while pc < ops.len() {
+                match run_op(&mut ltr, &lthdr, ops, pc) {
+                    RunResult::Ok(bytes) => {
+                        pc += bytes as usize;
+                    }
+                    RunResult::NewRow(bytes) => {
+                        pc += bytes as usize;
+                        row_cnt += 1;
+                        if addr < ltr.address {
+                            if row_cnt == 1 {
+                                return None;
+                            }
+                            ltr = saved_ltr.clone();
+                            break;
+                        }
+                        saved_ltr = ltr.clone();
+                    }
+                    RunResult::End | RunResult::Err => {
+                        break;
+                    }
+                }
+            }
+
+            if row_cnt == 0 {
+                continue;
+            }
+
+            let finfo = self.ctx.file_info(ltr.file_idx as usize);
+            let dirname = self.ctx.get_str(finfo.directory as usize);
+            let filename = self.ctx.get_str(finfo.filename as usize);
+            let path = format!("{}/{}", dirname, filename);
```

Comment:
Are you hard-coding operating system dependent (path separator) path logic here?